lua/telescope/previewers/buffer_previewer.lua

```diff
@@ -872,4 +867,35 @@ previewers.display_content = defaulter(function(_)
   }
 end, {})
 
+previewers.buffers = defaulter(function(opts)
+  opts = opts or {}
+  local cwd = opts.cwd or vim.loop.cwd()
+  return Previewer:new {
+    title = function()
+      return "Buffers"
+    end,
+    setup = function(_, status)
+      return { ["winid"] = status.preview_win }
+    end,
+    dyn_title = function(_, entry)
+      return Path:new(from_entry.path(entry, true)):normalize(cwd)
+    end,
+    preview_fn = function(self, entry, status)
+      if vim.api.nvim_buf_is_valid(entry.bufnr) then
+        vim.api.nvim_win_set_buf(status.preview_win, entry.bufnr)
+        vim.api.nvim_win_set_option(status.preview_win, "winhl", "Normal:TelescopePreviewNormal")
+        vim.api.nvim_win_set_option(status.preview_win, "signcolumn", "no")
+        vim.api.nvim_win_set_option(status.preview_win, "foldlevel", 100)
+        vim.api.nvim_win_set_option(status.preview_win, "wrap", false)
+        self.state.bufnr = entry.bufnr
+        if not entry.col then
+          local _, col = unpack(vim.api.nvim_win_get_cursor(status.preview_win))
```

Comment:
Current master sets `pcall(vim.api.nvim_buf_add_highlight, bufnr, ns_previewer, "TelescopePreviewLine", lnum - 1, 0, -1
` to indicate the current line, we should also do something similar here.

Reply:
Agree we should, but I have to think more about how to. `nvim_buf_add_highlight` cannot be used, since it's highlighting the buffer of course in previewer and background :laughing:

Reply:
i am kinda glad i skipped this issue, seems like a lot of work for terminal highlighting

Reply:
I think this is not impossible with `nvim_win_set_decoraton_provider` :laughing:

Reply:
Yeah, I've got it working with `nvim_win_set_decoration_provider`. I require Github Sponsors from anyone who upvoted the original issue :laughing: